apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp

```diff
@@ -0,0 +1,521 @@
+    _lineFovV->setVisible(newSelection == ProjectionIndeces::Planar);
+    _labelHeightOffset->setVisible(newSelection == ProjectionIndeces::Cylindrical);
+    _lineHeightOffset->setVisible(newSelection == ProjectionIndeces::Cylindrical);
+    _checkBoxSpoutOutput->setVisible(newSelection == ProjectionIndeces::Fisheye
+        || newSelection == ProjectionIndeces::Equirectangular);
+}
+
+void WindowControl::setDimensions(const QRectF& dimensions) {
+    _windowDims = dimensions;
+}
+
+void WindowControl::setWindowChangeCallback(
+                        std::function<void(int, int, const QRectF&)> cb)
+{
+    _windowChangeCallback = cb;
+}
+
+void WindowControl::setWebGuiChangeCallback(
+                        std::function<void(unsigned int)> cb)
+{
+    _windowGuiCheckCallback = cb;
+}
+
+void WindowControl::uncheckWebGuiOption() {
+    _checkBoxWebGui->setCheckState(Qt::Unchecked);
+}
+
+QRectF& WindowControl::dimensions() {
+    return _windowDims;
+}
+
+QLineEdit* WindowControl::lineEditSizeWidth() {
+    return _size_x;
+}
+
+QLineEdit* WindowControl::lineEditSizeHeight() {
+    return _size_y;
+}
+
+QLineEdit* WindowControl::lineEditSizeOffsetX() {
+    return _offset_x;
+}
+
+QLineEdit* WindowControl::lineEditSizeOffsetY() {
+    return _offset_y;
+}
+
+QCheckBox* WindowControl::checkBoxWindowDecor() {
+    return _checkBoxWindowDecor;
+}
+
+QCheckBox* WindowControl::checkBoxWebGui() {
+    return _checkBoxWebGui;
+}
+
+QCheckBox* WindowControl::checkBoxSpoutOutput() {
+    return _checkBoxSpoutOutput;
+}
+
+std::string WindowControl::windowName() {
+    return _windowName->text().toStdString();
+}
+
+sgct::ivec2 WindowControl::windowSize() {
+    return {
+        stoi(_size_x->text().toStdString()),
+        stoi(_size_y->text().toStdString())
+    };
+}
+
+sgct::ivec2 WindowControl::windowPos() {
+    return {
+        stoi(_offset_x->text().toStdString()),
+        stoi(_offset_y->text().toStdString())
+    };
+}
+
+bool WindowControl::isDecorated() {
+    return (_checkBoxWindowDecor->checkState() == Qt::Checked);
+}
+
+bool WindowControl::isGuiWindow() {
+    return (_checkBoxWebGui->checkState() == Qt::Checked);
+}
+
+bool WindowControl::isSpoutSelected() {
+    return (_checkBoxSpoutOutput->checkState() == Qt::Checked);
+}
+
+int WindowControl::projectionSelectedIndex() {
+    return _comboProjection->currentIndex();
+}
+
+int WindowControl::qualitySelectedIndex() {
+    return _comboQuality->currentIndex();
+}
+
+int WindowControl::qualitySelectedValue() {
+    return QualityValues[_comboQuality->currentIndex()];
+}
+
+float WindowControl::fovH() {
+    return _lineFovH->text().toFloat();
+}
+
+float WindowControl::fovV() {
+    return _lineFovV->text().toFloat();
+}
+
+float WindowControl::heightOffset() {
+    return _lineHeightOffset->text().toFloat();
+}
+
+unsigned int WindowControl::monitorNum() {
+    return _monIndex;
+}
+
+WindowControl::~WindowControl()
```

Comment:
Move the destructor up to just below the constructor and `{` needs to be on the asme line as the function name